File: python/src/skale_contracts/projects/skale_manager/skale_manager_project.py
from instance import Instance
from project import Project, ProjectMetadata
from .skale_manager_instance import SkaleManagerInstance


class SkaleManagerProject(Project):
    def __init__(self, network, metadata: ProjectMetadata) -> None:
        super().__init__(network, metadata)

    @property
    def github_repo(self):
        return 'https://github.com/skalenetwork/skale-manager/'

    def create_instance(self, address: str) -> Instance:
        return SkaleManagerInstance(self, address)

    def get_abi_filename(self, version: str):
        return f'skale-manager-{version}-abi.json'

File: typescript/base/src/projects/skale-manager/skaleManagerProject.ts
import { ContractAddressMap, MainContractAddress } from "../../domain/types";
import {
    SkaleManagerContract,
    SkaleManagerContractName,
    SkaleManagerDefaultTypesMap,
    SkaleManagerInstance
} from "./skaleManagerInstance";
import { Project } from "../../project";

export class SkaleManagerProject<ContractType> extends
    Project<ContractType, SkaleManagerContractName> {
    githubRepo = "https://github.com/skalenetwork/skale-manager/";

    mainContractName = SkaleManagerContract.SKALE_MANAGER;

    createInstance<
        TypesMap extends Record<SkaleManagerContractName, ContractType>=
            SkaleManagerDefaultTypesMap<ContractType>
    > (address: MainContractAddress | ContractAddressMap)
        : SkaleManagerInstance<ContractType, TypesMap> {
        return new SkaleManagerInstance<ContractType, TypesMap>(
            this,
            address
        );
    }

    getInstance<
        TypesMap extends Record<SkaleManagerContractName, ContractType>=
            SkaleManagerDefaultTypesMap<ContractType>
    > (
        target: string | MainContractAddress | ContractAddressMap
    ): SkaleManagerInstance<ContractType, TypesMap> {
        return super.getInstance<TypesMap>(
            target
        ) as SkaleManagerInstance<ContractType, TypesMap>;
    }

    getAbiFilename (version: string) {
        return `${this.metadata.name}-${version}-abi.json`;
    }
}
